packages/core/src/lib/message/version_0.ts

```diff
@@ -119,16 +123,17 @@ export class Encoder implements IEncoder {
  * pass to { @link @waku/interfaces!ISender.send } to automatically encode outgoing
  * messages.
  */
-export function createEncoder({
-  pubsubTopicShardInfo,
-  contentTopic,
-  ephemeral,
-  metaSetter
-}: EncoderOptions): Encoder {
+export function createEncoder(
+  { pubsubTopicShardInfo, contentTopic, ephemeral, metaSetter }: EncoderOptions,
+  autosharding = false,
+  clusterId = 0
+): Encoder {
```

Comment:
similar comment as above: we can move these two args into `EncoderOptions` if they're important for the encoder creation

Reply:
---
why do we need the `clusterId` flag? perhaps we can leverage the `shardInfo` arg and even remove the `autosharding` flag

Reply:
Updated in latest commit